source/dub/packagemanager.d

```diff
@@ -713,10 +714,12 @@ class PackageManager {
 		enforce(to_remove.length > 0, "No "~type.to!string()~" package found at "~path.toNativeString());
 
 		string[Version] removed;
-		foreach_reverse( i; to_remove ) {
+		foreach (i; to_remove)
 			removed[(*packs)[i].version_] = (*packs)[i].name;
-			*packs = (*packs)[0 .. i] ~ (*packs)[i+1 .. $];
-		}
+
+		*packs = (*packs).enumerate
+			.filter!(en => to_remove.canFind(en.index))
```

Comment:
```suggestion
			.filter!(en => !to_remove.canFind(en.index))
```

should this not be inverted?

Reply:
`filter` always confused me.

Reply:
can we add a test case for this? before the CI failed on something which didn't exactly look like a direct cause by this (only really through a lot of other things)

the whole pointer usage here makes this code kinda more complicated to read and reason about as there might be slices of the data used elsewhere in other code.

Reply:
> can we add a test case for this? before the CI failed on something which didn't exactly look like a direct cause by this (only really through a lot of other things)

I tried writing a test for it, but ended up with the same test code, because one will just do `add-local`, something idempotent (describe, list), and then `remove-local`, which is what `version-spec.sh` does. If anything, we should probably rename `version-spec.sh` because it does test a lot of things.

Reply:
> the whole pointer usage here makes this code kinda more complicated to read and reason about as there might be slices of the data used elsewhere in other code.

Agreed, will try to remove it in the future.